data_steward/cdr_cleaner/manual_cleaning_rules/update_questiona_answers_not_mapped_to_omop.py

```diff
@@ -0,0 +1,167 @@
+        FROM (
+            SELECT
+            SUBSTR(pmi_code,1,50) AS short_pmi_code,
+            short_code
+            FROM
+                `{project}.{sandbox}.{old_map}`
+            WHERE
+                type='Question' )
+        LEFT JOIN
+            `{project}.{dataset}.concept` source_c
+        ON
+            (short_code=concept_code)
+        JOIN
+            `{project}.{dataset}.concept_relationship` c_r  
+        ON
+            (source_c.concept_id=c_r.concept_id_1)
+        JOIN
+            `{project}.{dataset}.concept` standard_c
+        ON
+            (standard_c.concept_id=c_r.concept_id_2)
+        WHERE
+            source_c.vocabulary_id='PPI'
+            AND c_r.relationship_id LIKE 'Maps to%'
+        ) map
+    WHERE
+        map.observation_source_value=obs.observation_source_value"""
+
+UPDATE_ANSWERS_MAP_QUERY = """
+    UPDATE
+        `{project}.{dataset}.observation` obs
+    SET
+        value_as_concept_id=new_value_as_concept_id,
+        value_source_concept_id=new_value_source_concept_id
+    FROM (
+        SELECT
+        DISTINCT short_pmi_code AS value_source_value,
+        source_c.concept_id AS new_value_source_concept_id,
+        FIRST_VALUE(standard_c.concept_id) OVER (PARTITION BY source_c.concept_id ORDER BY c_r.relationship_id DESC ) AS new_value_as_concept_id
+        FROM (
+            SELECT
+            SUBSTR(pmi_code,1,50) AS short_pmi_code,
+            short_code
+            FROM
+                `{project}.{sandbox}.{old_map}`
+            WHERE
+                type='Answer' )
+        LEFT JOIN
+            `{project}.{dataset}.concept` source_c
+        ON
+            (short_code=concept_code)
+        JOIN
+            `{project}.{dataset}.concept_relationship` c_r
+        ON
+            (source_c.concept_id=c_r.concept_id_1)
+        JOIN
+            `{project}.{dataset}.concept` standard_c
+        ON
+            (standard_c.concept_id=c_r.concept_id_2)
+        WHERE
+            source_c.vocabulary_id='PPI'
+        AND c_r.relationship_id LIKE 'Maps to%') map
+    WHERE
+        map.value_source_value=obs.value_source_value"""
+
+
+def get_update_questions_answers_not_mapped_to_omop(project_id, dataset_id, sandbox_dataset_id):
+    """
+
+    :param project_id: Name of the project
+    :param dataset_id: Name of the dataset where the queries should be run
+    :param sandbox_dataset_id: Name of the sandbox dataset
+    :return:
+    """
+
+    bq_utils.load_table_from_csv(project_id=project_id,
+                                 dataset_id=sandbox_dataset_id,
+                                 table_name=OLD_MAP_SHORT_CODES_TABLE,
+                                 csv_path=None,
+                                 fields=OLD_MAP_SHORT_CODES_TABLE_FIELDS)
+
+    queries_list = []
+
+    # Update concept_ids to questions using OLD_MAP_SHORT_CODES_TABLE.
+    query = dict()
```

Comment:
If you use one query instead of two separate queries, this can become
```
for concept_type in ['Question', 'Answer']:
    query = {}
    query[cdr_consts.QUERY] = UPDATE_MAP_QUERY.format(
        dataset=dataset_id,
        project=project_id,
        old_map=OLD_MAP_SHORT_CODES_TABLE,
        sandbox=sandbox_dataset_id
    )
queries_list.append(query)
```

Reply:
No longer relevant.